### PR TITLE
Replace all user-facing "Stash" with "Bitbucket Server"

### PR DESCRIPTION
**Pull request by darxriggs**
The current wording & descriptions still contains **Stash** but also a little bit of **Bitbucket**. This is quite confusing for users. Especially as Stash is not well known anymore and Bitbucket could mean "Bitbucket Server" or "Bitbucket Cloud".

Therefore this pull-request replaces all user-facing "Stash" with "Bitbucket Server" to make things more clear.

Details (from the commit comment).
* Update comments and examples in the code too.
* Don't touch methods and properties as this would break existing configurations.
* Also cleanup README and update old links.

### CODE REVIEW

**Review by scaytrase — COMMENTED** [review]
Althought I like this PR in general this one should be carefully tested before merging, since we broke the plugin distribution by replacing plugin link in pom.xml once ago, so I don't really know which things could break it again.

I'll go throught it carefully as soon as I can

**Comment by scaytrase** [comment]
I hope this one would be easier to rebase, so I merged #201 first. @darxriggs rebase it please

**Comment by darxriggs** [comment]
@scaytrase rebase done.

File: readme.md [hunk 1]
```diff
@@ -1,77 +1,74 @@
...
+commit logs. Clicking on such an icon will take the user to the specific build.
 
 Requirements
 ============
 
-* **[Stash][] 2.1** or newer. This plugin uses the Atlassian 
-[Stash Build REST API][] which was introduced with Stash 2.1. 
-* **Jenkins 1.498** or newer
+* **[Stash][] 2.1** or newer / **[Bitbucket Server][] 4.0** or newer
+* **Jenkins 1.625.3** or newer
```
Comment (scaytrase) [inline readme.md:12]: we actually forgot to mention new requirements here, but this is not a problem of this PR

**Review by scaytrase — APPROVED** [review]

**Comment by scaytrase** [comment]
thank you @darxriggs

**Pull request merged**